### ISSUE TITLE
Empty results on univariate dataset scripts

### ISSUE THREAD

**Issue by stefanosh** [decisionintelligence/TFB#14]
Hello, thanks a lot for your effort into this comprehensive forecasting benchmark.

I tried to run for univariate dataset "weather_dataset_3.csv" (downloaded from your google drive link) but I get empty results. For example,  running

 ``
python ./scripts/run_benchmark.py --config-path "fixed_forecast_config_daily.json" --data-name-list "weather_dataset_3.csv"  --model-name  "time_series_library.Triformer"  --model-hyper-params "{\"d_model\":16,\"d_ff\":16,\"factor\":3}" --adapter "transformer_adapter" --save-path "daily"   --gpus 0 1  --num-workers 1 --timeout 60000
``

trains successfully the model with no errors and creates the zip file. However, when unzipping the test .csv file it is empty without metric results (see figures below).

For rolling_forecast as below, I get all the results.
``
python ./scripts/run_benchmark.py --config-path "rolling_forecast_config.json" --data-name-list "Weather.csv" --strategy-args '{"horizon":96}' --model-name "time_series_library.Linear" --model-hyper-params '{"d_ff": 64, "d_model": 32, "horizon": 96, "seq_len": 512}' --adapter "transformer_adapter"  --gpus 0  --num-workers 1  --timeout 60000  --save-path "Weather/Linear"
``

1) What can be the cause for this?
2) Besides this issue, what are your recommendations/guidelines for running the benchmark on a custom dataset? e.g., a) data format, b) config file update, c) hyperparameter tuning

Thanks.

![image](https://github.com/decisionintelligence/TFB/assets/9021406/8f79bf59-32c2-4dd3-9e8e-a466f545145f)

![image](https://github.com/decisionintelligence/TFB/assets/9021406/eb831c23-45f3-43c8-85c7-9061e652fddd)

**Comment by qiu69** [comment]
python ./scripts/run_benchmark.py --config-path "fixed_forecast_config_daily.json" --data-name-list "weather_dataset_3.csv"  --model-name  "time_series_library.Triformer"  --model-hyper-params "{\"d_model\":16,\"d_ff\":16,\"factor\":3}" --adapter "transformer_adapter" --save-path "daily"   --gpus 0 1  --num-workers 1 --timeout 60000
Thank you for your attention and question. I ran the same command, but I got the result as follows:
<img width="1428" alt="image" src="https://github.com/decisionintelligence/TFB/assets/82441982/f9a4cb7a-c125-4c30-94fa-dd865035cbd8">
<img width="1412" alt="image" src="https://github.com/decisionintelligence/TFB/assets/82441982/adfc7cc5-e99c-4d87-85ca-3d1784f82366">

**Comment by qiu69** [comment]
For the second question, we will provide a dedicated tutorial later on.
You can now follow the following tutorial to use your own dataset.
Step 1: Convert your own dataset to our format, which is a 3-column long table. You can open a multivariate dataset or a univariate dataset to view it.
Step 2: Add your dataset information to the FORECAST-META.csv file. Please note that you only need to fill in attributes such as filename, freq, if_univariant, size, and length. Please fill in the size as large, and the length is the original length of your dataset, not the length of the dataset converted to our format.
Step 3: Specify your own dataset name in the command, such as: python ./scripts/run_benchmark.py --config-path "fixed_forecast_config_daily.json" --data-name-list "your dataset name" --model-name "time_series_library.Triformer" --model-hyper-params "{"d_model":16,"d_ff":16,"factor":3}" --adapter "transformer_adapter" --save-path "daily" --gpus 0 1 --num-workers 1 --timeout 60000

**Comment by qiu69** [comment]
For the second question, we will provide a dedicated tutorial later on.
You can now follow the following tutorial to use your own dataset.
Step 1: Convert your own dataset to our format, which is a 3-column long table. You can open a multivariate dataset or a univariate dataset to view it.
Step 2: Add your dataset information to the FORECAST-META.csv file. Please note that you only need to fill in attributes such as filename, freq, if_univariant, size, and length. Please fill in the size as large, and the length is the original length of your dataset, not the length of the dataset converted to our format.
Step 3: Specify your own dataset name in the command, such as: python ./scripts/run_benchmark.py --config-path "fixed_forecast_config_daily.json" --data-name-list "your dataset name" --model-name "time_series_library.Triformer" --model-hyper-params "{"d_model":16,"d_ff":16,"factor":3}" --adapter "transformer_adapter" --save-path "daily" --gpus 0 1 --num-workers 1 --timeout 60000

**Comment by qiu69** [comment]
a) If you want to compare the performance with the univariate algorithm in the paper, you should follow the configuration in the existing configuration file and not modify the forecasting horizon and look back window size. The hyperparameters of the algorithms themselves, such as training epochs and lr, can be modified.
b) Whether to perform rolling forecast is another usage scenario and also reflects another ability of the algorithm. You can decide according to your needs.

**Comment by qiu69** [comment]
For each algorithm on a dataset, we ran multiple sets of hyperparameters and reported the performance results under the best parameters in the paper. The univariate scripts provide examples of batch evaluating univariate datasets. 
We suggest that you use the settings in the config file, including seed. You can also write your own config file according to your needs

**Comment by qiu69** [comment]
Hi, If you have any further questions, please feel free to report them to us. If the problem has been resolved, please close the issue.
